rts/Rendering/Fonts/CFontTexture.cpp

```diff
@@ -898,36 +1156,54 @@ void CFontTexture::LoadGlyph(std::shared_ptr<FontFace>& f, char32_t ch, unsigned
 	if (glyph.advance == 0 && glyph.size.w > 0)
 		glyph.advance = glyph.size.w;
 
-	const int width  = slot->bitmap.width;
-	const int height = slot->bitmap.rows;
+	int width  = slot->bitmap.width;
+	int height = slot->bitmap.rows;
 	const int olSize = 2 * outlineSize;
+	const int channels = slot->bitmap.pixel_mode == FT_PIXEL_MODE_BGRA ? 4 : 1;
 
-	if (width <= 0 || height <= 0)
-		return;
-
-	if (slot->bitmap.pixel_mode != FT_PIXEL_MODE_GRAY) {
+	if (slot->bitmap.pixel_mode != FT_PIXEL_MODE_GRAY && slot->bitmap.pixel_mode != FT_PIXEL_MODE_BGRA) {
 		LOG_L(L_ERROR, "invalid pixeldata mode");
 		return;
 	}
 
-	if (slot->bitmap.pitch != width) {
-		LOG_L(L_ERROR, "invalid pitch");
+	if (slot->bitmap.pitch != width*channels) {
+		LOG_L(L_ERROR, "invalid pitch %d (width %d channels %d)", slot->bitmap.pitch, width, channels);
```

Comment:
Hmm two concerns:

* pitch | its absolute value is the number of bytes per bitmap line; it can be either positive or negative depending on the bitmap's vertical orientation;
* To speed up memory access, pitch is in most cases a multiple of 16bit, 32bit, or even 64bit. It often happens that the pitch is thus larger than the necessary bits (or bytes) for a bitmap or pixmap row; in such cases, unused bits (or bytes) are at the very right (i.e., the end) of a row.

https://freetype.org/freetype2/docs/glyphs/glyphs-7.html

Reply:
@lhog, you're right, but also I think the test we're doing here holds, since in those cases it will complain about the invalid pitch.

Maybe in that case we could bail out instead of just continuing, but just continue is the same logic we had here before, so I just adapted the test to also work on the RGBA case similarly, but otherwise didn't change it. As I understood, the test is an alert in case we even encounter that issue, but it's also an issue that didn't pop up yet.

Personally, would rather keep as-is and later work on this to make it bail out if this happens, or even support the negative pitch or `pitch!=width*channels` cases.